clang/test/SemaTemplate/alias-template-deprecated.cpp

```diff
@@ -0,0 +1,58 @@
+// RUN: %clang_cc1 -std=c++11 -fsyntax-only -verify %s
+//
+// This test checks that a deprecated attribute on an alias
+// template triggers a warning diagnostic when it is used.
+
+template <typename T>
+struct NoAttr {
+  void foo() {}
+};
+
+// expected-note@+2 5{{'UsingWithAttr' has been explicitly marked deprecated here}}
+template <typename T>
+using UsingWithAttr __attribute__((deprecated)) = NoAttr<T>;
```

Comment:
I recommend adding a test that uses the C++11 `[[deprecated]]` attribute syntax.

Reply:
Thank you @tahonermann!  Done.